src/lib.rs

```diff
@@ -2831,6 +2831,47 @@ pub trait Itertools : Iterator {
         self.for_each(|item| *counts.entry(item).or_default() += 1);
         counts
     }
+
+    /// An alternative to `fold()` - `fold_mut()` also applies a function
+    /// producing a single value. The main difference is that the closure
+    /// passed to `fold_mut()` can accept a `&mut` to the accumulator instead
+    /// of requiring ownership of the accumulator. This can simplify some closures
+    /// that might otherwise awkwardly be forced to return the accumulator:
+    /// ```
+    /// let evens = [1, 2, 3, 4, 5, 6].iter().fold(Vec::new(), |mut evens, num| {
+    ///   if num % 2 == 0 {
+    ///     evens.push(num);
+    ///   }
+    ///   evens // potentially awkward return
+    /// });
+    /// ```
+    ///
+    /// `fold_mut()` may also be more performant in situations where the
+    /// accumulator is "large" as passing it by `&mut` can be cheaper than
+    /// transferring ownership of the accumulator.
+    ///
+    /// # Examples
+    /// ```
+    /// # use itertools::Itertools;
+    /// let evens = [1, 2, 3, 4, 5, 6].iter().fold_mut(Vec::new(), |evens, &num| {
```

Comment:
Is this example too lame? I was going to use the example of counting but ... there's `.counts` right above this so that seemed worse haha